lib/domain/usecase/room/chat_get_pinned_events_interactor.dart

```diff
@@ -8,14 +8,32 @@ import 'package:matrix/matrix.dart';
 
 class ChatGetPinnedEventsInteractor {
   Stream<Either<Failure, Success>> execute({
-    required Room room,
+    required String roomId,
+    required Client client,
+    bool isInitial = false,
   }) async* {
     Logs().d(
-      "ChatGetPinnedEventsInteractor()::execute()::roomId: ${room.id}",
+      "ChatGetPinnedEventsInteractor()::execute()::roomId: $roomId",
     );
-    yield Right(ChatGetPinnedEventsLoading());
+    if (isInitial) {
+      yield Right(ChatGetPinnedEventsLoading());
+    }
     try {
+      final room = client.getRoomById(roomId);
+      if (room == null) {
+        Logs().d(
+          "ChatGetPinnedEventsInteractor()::execute(): Room is Null",
+        );
+        yield Left(ChatGetPinnedEventsRoomIsNull());
```

Comment:
WDYT `ChatGetPinnedEventsRoomIsNull` should be  `ChatGetPinnedEventsFailure` with `nullRoomException`

Reply:
I identify this as a specific error, which makes it easier for us to debug later. I don't care about `Exception`